pkg/provision/engine.go

```diff
@@ -952,3 +952,21 @@ func (e *NativeEngine) DecommissionCached(id string, reason string) error {
 
 	return err
 }
+
+// GetWorkloadStatus get workload status
+func (e *NativeEngine) GetWorkloadStatus(id string) (gridtypes.ResultState, error) {
+	globalID := gridtypes.WorkloadID(id)
+	twin, dlID, name, err := globalID.Parts()
+	if err != nil {
+		return "", err
+	}
+	wl, err := e.storage.Current(twin, dlID, name)
+	if err != nil {
+		if e.storage.IsNotFoundError(err) {
```

Comment:
This should not be on the insterface. when i said `storage.IsNotFound()` i meant on the storage package so you can call this function as `storage.IsNotFound(err)` directly !